temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -203,22 +204,23 @@ private void handleEventsBatch(List<HistoryEvent> events, boolean hasNextEvent)
     }
   }
 
-  private InternalWorkflowTaskException createEventProcessingException(
-      RuntimeException e, HistoryEvent event) {
-    return new InternalWorkflowTaskException(
-        "Failure handling event "
-            + event.getEventId()
-            + " of '"
-            + event.getEventType()
-            + "' type. IsReplaying="
-            + this.isReplaying()
-            + ", PreviousStartedEventId="
-            + this.getLastStartedEventId()
-            + ", workflowTaskStartedEventId="
-            + this.workflowTaskStartedEventId
-            + ", Currently Processing StartedEventId="
-            + this.currentStartedEventId,
-        unwrap(e));
+  private RuntimeException createEventProcessingException(RuntimeException e, HistoryEvent event) {
+    Throwable ex = unwrap(e);
+    if (ex instanceof NonDeterministicException) {
+      // just appending the message in front of an existing message, saving the original stacktrace
+      NonDeterministicException modifiedException =
```

Comment:
What if `ex` has a cause? This swallows it.

Reply:
It doesn't now, I create all NonDeterministicExdeption without cause, but a good catch! I will fix it.